### PR TITLE
Update ExMachina, Ecto and other dependencies. Fix Elixir 1.4 warnings

### PR DESCRIPTION
**Pull request by paulcsmith**
* Test out the latest ExMachina. It handles casting automatically now.
* Upgrade Ecto to 2.1
* Switch to native Elixir DateTime objects (available in 1.4)
* Upgrade a few other deps. See `mix.lock`

### CODE REVIEW

**Comment by paulcsmith** [comment]
@BlakeWilliams I think that last commit broke a few tests on master. Could you take a look when you have a sec?

**Comment by BlakeWilliams** [comment]
Whoops, yep!

**Comment by paulcsmith** [comment]
It may be helpful to review this commit by commit since the second one mostly fixes parens and makes it hard to see what changed while upgrading

File: mix.exs [hunk 1]
```diff
@@ -49,17 +49,17 @@ defmodule Constable.Mixfile do
...
-      {:cors_plug, "~> 0.1"},
+      {:cors_plug, "~> 1.2"},
       {:cowboy, "~> 1.0"},
       {:dialyxir, "~> 0.3", only: [:dev]},
       {:earmark, github: "pragdave/earmark"},
-      {:ecto, "~> 2.0"},
+      {:ecto, "~> 2.1"},
       {:envy, "~> 0.0.1"},
-      {:ex_machina, "~> 1.0"},
+      {:ex_machina, "~> 1.0", github: "thoughtbot/ex_machina"},
```
Comment (BlakeWilliams) [inline mix.exs]: Why did this change to github?

File: lib/constable/time.ex [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
   def days_ago(count) do
-    # Date.now |> Date.shift(days: -count) |> to_ecto
-    GoodTimes.days_ago(count) |> to_ecto
+    GoodTimes.days_ago(count) |> cast!
   end
 
-  def to_ecto(date) do
-    date |> Ecto.DateTime.cast!
+  def cast!(time) do
+    time |> NaiveDateTime.from_erl! |> DateTime.from_naive!("Etc/UTC")
```
Comment (BlakeWilliams) [inline lib/constable/time.ex]: Nice!

File: mix.exs (hunk 1 above)
Comment (paulcsmith) [inline mix.exs]: Because it's not released to hex now. When it is we'll switch it back over

Comment (BlakeWilliams) [inline mix.exs]: Gotcha!

**Pull request merged**